scipy/integrate/_ivp/ivp.py

```diff
@@ -492,6 +537,10 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
     events, is_terminal, event_dir = prepare_events(events)
 
     if events is not None:
+        if args is not None:
+            # Wrap user functions in lambdas to hide the additional parameters.
```

Comment:
Would be good to explain why `event=event:` is needed.

Reply:
I added a comment (here and up by the `fun` lambda wrapper about using the keyword argument to keep the original function in scope.